oak_runtime/src/node/http/client.rs

```diff
@@ -0,0 +1,327 @@
+    http_client: HyperClient,
+    /// The authority corresponding to this client, or none in case of a public client.
+    authority: Option<String>,
+    /// Downgrading privilege of this client.
+    node_privilege: NodePrivilege,
+}
+
+/// Oak Node implementation for the HTTP client pseudo-Node.
+impl Node for HttpClientNode {
+    fn node_type(&self) -> &'static str {
+        "HTTP-client"
+    }
+
+    fn run(
+        mut self: Box<Self>,
+        runtime: RuntimeProxy,
+        handle: Handle,
+        notify_receiver: oneshot::Receiver<()>,
+    ) {
+        // Create an Async runtime for executing futures.
+        // https://docs.rs/tokio/
+        // TODO(#1280): Use a single shared tokio runtime, instead of creating a new one here
+        let mut async_runtime = create_async_runtime();
+
+        // Listen to incoming HTTP invocations.
+        info!(
+            "{}: Starting HTTP client pseudo-Node thread",
+            self.node_name
+        );
+        async_runtime.block_on(futures_util::future::select(
+            Box::pin(self.handle_loop(runtime, handle)),
+            notify_receiver,
+        ));
+        info!("{}: Exiting HTTP client pseudo-Node thread", self.node_name);
+    }
+
+    fn get_privilege(&self) -> NodePrivilege {
+        self.node_privilege.clone()
+    }
+}
+
+fn get_privilege(authority: String) -> NodePrivilege {
+    if authority.is_empty() {
+        NodePrivilege::default()
+    } else {
+        NodePrivilege::new(
+            hashset! { oak_abi::label::tls_endpoint_tag(&authority.as_str()) },
+            hashset! { oak_abi::label::tls_endpoint_tag(&authority.as_str()) },
+        )
+    }
+}
+
+/// Checks the validity of the configuration:
+/// - A non-empty TLS root certificate is provided
+/// - If the scheme is HTTPS, a non-empty authority is specified in the config
+fn validate_config(config: &HttpClientConfiguration, empty_root_ca: bool) -> bool {
+    if empty_root_ca {
+        error!("HTTP client pseudo-node: No root TLS certificate provided.");
+    }
+    if config.scheme == UriScheme::Https as i32 && config.authority.is_empty() {
+        error!("HTTP client pseudo-node: No authority specified.");
+    }
+    !empty_root_ca && (config.scheme != UriScheme::Https as i32 || !config.authority.is_empty())
+}
+
+impl HttpClientNode {
+    /// Creates a new `HttpClientNode` that can either handle HTTP requests or HTTPS requests. To
+    /// handle HTTPS requests a non-empty authority in the `config` is required. This configuration
+    /// must also match the label with which this node is being created. If the node is configured
+    /// to handle URIs with the `HTTP` scheme, then it should be labelled public. In this case,
+    /// the authority could be empty, meaning that the node can handle all HTTP requests. In
+    /// addition, a non-empty `root_ca` is always required.
+    pub fn new(node_name: &str, config: HttpClientConfiguration, root_ca: Vec<u8>) -> Option<Self> {
```

Comment:
I guess this could be fixed consistently for all nodes in a separate PR. If there is a consensus about this convention, I can create an issue for it :)